src/FlashMessage/FlashManager.php

```diff
@@ -18,8 +18,10 @@
 
 /**
  * @author Vincent Composieux <composieux@ekino.com>
+ *
+ * NEXT_MAJOR: remove StatusClassRendererInterface
  */
-final class FlashManager implements StatusClassRendererInterface
```

Comment:
Removing an interface here is a BC break.

Reply:
FlashManager was tagged by `sonata.status.render` (had to implement `StatusClassRendererInterface`). This will allowed using `sonata_status_class` twig filter. After upgrade and add return types Flash Message cannot use it becouse it is string and StatusClassRenderer require object. Technicaly this is BC-break but can we not implement in this case? 

BTW, remove tag from FlashManager service is BC-break too, anyway remove something what can run is BC-break?

https://github.com/sonata-project/twig-extensions/blob/1.x/docs/reference/status_helper.rst

Reply:
Usually we would do something like
FlashManager implements StatusClassRendererInterface, FlashManagerInterface

But for what I understand the current interface is not compatible. If we have no other option, it could be acceptable, as long as we have an upgrade note.

Reply:
__StatusClassRenderer__ is feature for getting css class from object status (expamle: success for correct payment).
__FlashManager__ is service for working with flash message witch use __StatusClassRenderer__.

StatusClassRendererInterface is require for getting status class from object. like:
```twig
{{ order|sonata_status_class('payment', 'danger') }}
{# from function statusClass(object $object, $statusType = null, string $default = ''): string #}
```
This object are register by DI (tag: `sonata.status.render`) some example:
https://github.com/sonata-project/twig-extensions/blob/1.x/docs/reference/status_helper.rst
https://github.com/sonata-project/ecommerce/blob/3.x/src/Component/Order/OrderStatusRenderer.php

FlashManager and render.html.twig use this status class in wrong way now. Previus API will allow for that but with type hints is is broken:
```twig
{# current bad way #}
{{ type|sonata_status_class }}  // type is string like: 'success' or 'sonata_user_success'

{# good way #}
{{ flashManager|sonata_status_class(type, 'default_class' }}  

{# my solution #}
{{ type|sonata_flashmessage_class('default_class' }}  
{# we can add too: {{ sonata_flashmessage_class(type, 'default_class' }} #}  
```

Good way require inject flashManager to templates and make little confusing when you get status class for object and for flash. 

My idea is silmply:
sonata_flashmessage_* - for flash
sonata_status_* - for object

Reply:
@core @VincentLanglet I will keep this interface, fix working for this methods and mark they as deprecated.